contracts/tokens/SecurityToken.sol

```diff
@@ -115,22 +117,22 @@ contract SecurityToken is ISecurityToken {
      * @param _decimals Decimals for the securityToken
      * @param _granularity granular level of the token
      * @param _tokenDetails Details of the token that are stored off-chain (IPFS hash)
-     * @param _securityTokenRegistry Contract address of the security token registry
+     * @param _STR_Address Contract address of the security token registry
      */
     constructor (
         string _name,
         string _symbol,
         uint8 _decimals,
         uint256 _granularity,
         string _tokenDetails,
-        address _securityTokenRegistry
+        address _STR_Address
     )
     public
     DetailedERC20(_name, _symbol, _decimals)
     {
         //When it is created, the owner is the STR
-        moduleRegistry = ISecurityTokenRegistry(_securityTokenRegistry).moduleRegistry();
-        polyToken = ERC20(ISecurityTokenRegistry(_securityTokenRegistry).polyAddress());
+        STR_Address = _STR_Address;
+        polyToken = ERC20(IRegistry(_STR_Address).getAddress("POLY_Address"));
```

Comment:
Rather than storing `polyToken` can we just look it up via the IRegistry(_STR_Address).getAdress("POLY_Address")) when we need it?

Reply:
What is the advantage in doing so? Is it because of the gas difference between memory vs storage?